File: parasmittal/Day 17/XOR Queries of a Subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
     vector<int> um(arr.size(),0);
     um[0]=arr[0];
     for(int i=1;i<arr.size();i++)
         um[i]=um[i-1]^arr[i];
      vector<int> ans;
      for(int i=0;i<queries.size();i++){
          int l=queries[i][0];
          int r=queries[i][1];
          int val=um[r];
          if(l!=0)
            val^=um[l-1];
          ans.push_back(val);
      
      }
        return ans;
    }
};

File: parnamondal/Week -4/Day -1/XOR Queries of a subarray.py
# Time Complexity - 0(32N)
# Space Complexity - O(N)

class Solution:
    def xorQueries(self, arr: List[int], qu: List[List[int]]) -> List[int]:
        n=len(arr)
        xorarr=[arr[0]]
        xor=arr[0]
        for i in arr[1:]:
            xor^=i 
            xorarr.append(xor)
        ans=[]
        for i in qu:
            l,r=i
            if l==r:
                ans.append(arr[l])
            elif l==0:
                ans.append(xorarr[r])
            else:
                ans.append(xorarr[r]^xorarr[l-1])
        return ans
                